javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -345,6 +345,18 @@ private int filter(Predicate<? super T> predicate, Object results, int index, T
         return index;
     }
 
+    Object[] toArray() {
+        final Object[] results = new Object[length];
+        visit((index, leaf, start, end) -> {
+            final int copied = end - start;
+            for (int i = start; i < end; i++) {
+                results[index + i] = type.getAt(leaf, i);
```

Comment:
> All types could benefit if we change Value.toArray() instead

Great idea, applied your suggestion and it's still faster than before (see measurements above), and only slightly slower that the previous impl (~10%)!